tests/spec/webpack-plugin/webpack-plugin.spec.js

```diff
@@ -0,0 +1,528 @@
+'use strict';
+const expect = require('chai').expect;
+const utils = require('../../lib/test-utils');
+const FilerWebpackPlugin = require('../../../src/webpack-plugin/plugin');
+
+function createNMFResolveCompilerObject(resolveData) {
+
+  const normalModuleFactory_resolve_tap = utils.createMockFn(function (name, callback) {
+    callback(resolveData);
+  });
+
+  const normalModuleFactory = {
+    hooks: {
+      resolve: {
+        tap: normalModuleFactory_resolve_tap
+      }
+    }
+  };
+
+  const normalModuleFactory_tap = utils.createMockFn(function (name, callback) {
+    callback(normalModuleFactory);
+  });
+
+  const compiler = {
+    hooks: {
+      normalModuleFactory: {
+        tap: normalModuleFactory_tap,
+      }
+    }
+  };
+
+  return {
+    compiler,
+    normalModuleFactory_tap,
+    normalModuleFactory_resolve_tap,
+  };
+}
+
+describe('path shim', () => {
+  it('should instantiate the plugin with valid options', () => {
+    // Arrange
+    const options = {
+      filerDir: '/filer',
+      shimsDir: '/shims',
+      fsProviderDir: '/fsProvider',
+      shimFs: false,
+      shimPath: false,
+      fsProvider: 'custom',
+    };
+
+    // Act
+    const plugin = new FilerWebpackPlugin(options);
+
+    // Assert
+    expect(plugin.options).to.not.be.undefined;
+
+    expect(plugin.options.filerDir).to.equal(options.filerDir);
+    expect(plugin.options.shimsDir).to.equal(options.shimsDir);
+    expect(plugin.options.fsProviderDir).to.equal(options.fsProviderDir);
+    expect(plugin.options.shimFs).to.equal(options.shimFs);
+    expect(plugin.options.shimPath).to.equal(options.shimPath);
+    expect(plugin.options.fsProvider).to.equal(options.fsProvider);
+  });
+
+  it('should instantiate the plugin with default options', () => {
+    // Act
+    const plugin = new FilerWebpackPlugin();
+
+    // Assert
+    expect(plugin.options).to.not.be.undefined;
+
+    expect(plugin.options.filerDir).to.equal('/node_modules/filer');
+    expect(plugin.options.shimsDir).to.equal('/node_modules/filer/shims');
+    expect(plugin.options.fsProviderDir).to.equal('/node_modules/filer/shims/providers');
+    expect(plugin.options.shimFs).to.equal(true);
+    expect(plugin.options.shimPath).to.equal(true);
+    expect(plugin.options.fsProvider).to.equal('default');
+  });
+
+  it('should instantiate the plugin with invalid options', () => {
```

Comment:
`should throw if provided invalid options` is more clear I think

Reply:
Yeah, good shout 👍

Reply:
Done